addons/sale_margin/models/sale_order.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from openerp import api, fields, models
+import openerp.addons.decimal_precision as dp
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    margin = fields.Float(compute='_product_margin', digits=dp.get_precision('Product Price'), store=True)
+    purchase_price = fields.Float(string='Cost Price', digits=dp.get_precision('Product Price'))
+
+    @api.multi
+    @api.onchange('product_id', 'product_uom_qty')
+    def product_id_change_margin(self):
+        for line in self:
```

Comment:
In this particular case, the `@api.multi` is indeed useless. In other cases such as here: https://github.com/odoo/odoo/blob/9.0/addons/sale/sale.py#L177 that was necessary since the method is manually called from some old-api code (I don't remember the exact reason, though).

Anyway the sale_* modules definitely need a cleaning after the major changes that were done.